src/Amazon.AspNetCore.DataProtection.SSM/SSMXmlRepository.cs

```diff
@@ -197,6 +209,96 @@ private async Task StoreElementAsync(XElement element, string friendlyName)
             }
         }
 
+#if NET9_0_OR_GREATER
+        /// <inheritdoc/>
+        public bool DeleteElements(Action<IReadOnlyCollection<IDeletableElement>> chooseElements)
+        {
+            return Task.Run(() => DeleteElementsAsync(chooseElements)).GetAwaiter().GetResult();
+        }
+
+        private async Task<bool> DeleteElementsAsync(Action<IReadOnlyCollection<IDeletableElement>> chooseElements)
+        {
+            if (chooseElements == null)
+            {
+                throw new ArgumentNullException(nameof(chooseElements));
+            }
+
+            var deletableElements = new List<DeletableElement>();
+            var request = new GetParametersByPathRequest
+            {
+                Path = _parameterNamePrefix,
+                WithDecryption = true
+            };
+            GetParametersByPathResponse response = null;
+
+            do
+            {
+                request.NextToken = response?.NextToken;
+                try
+                {
+                    response = await _ssmClient.GetParametersByPathAsync(request).ConfigureAwait(false);
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError(
+                        e,
+                        "Error calling SSM to get parameters starting with {ParameterNamePrefix}: {ExceptionMessage}",
+                        _parameterNamePrefix,
+                        e.Message);
+
+                    throw;
+                }
+
+                foreach (var parameter in response.Parameters)
```

Comment:
For V4 we need a null check for `response.Parameters`.

Reply:
Fixed